crates/sparrow-compiler/src/functions/collection.rs

```diff
@@ -14,9 +14,9 @@ pub(super) fn register(registry: &mut Registry) {
         .set_internal();
 
     registry
-        .register("collect<T: any>(const max: i64, input: T, window: window = null) -> list<T>")
+        .register("collect<T: any>(input: T, const max: i64, const min: i64 = 0, window: window = null) -> list<T>")
```

Comment:
What's the benefit of making this part of `collect`? Are there ways it could be implemented more efficiently as a result? Are there ways it is more expressive to the user? If the former, then it could be part of the instruction, but not necessarily the function. If the latter, then it could be part of the Python builder, but not necessarily required.

The alternative would be to filter after `collect` using existing functionality:
```python
collected = collect(input, max)
collected.if_(collected.len_() >= min)
```

One benefit of that is that it is trivial to share the collecting across multiple cases with different min (we could even do different max that way). That said, if we wish to do this fully generally, we'd likely need to have an optimization that does something like that, so it doesn't hurt to have this be part of the function.

Thoughts? Worth discussing tomorrow?

Reply:
This is a good point. Let's discuss tomorrow

Reply:
As discussed, one major benefit is that we avoid cloning values from `0..min`. For larger lists and min values, this can be a big performance gain. 

Especially if `collect` is used to implement lag, we'll consistently have the signature 
```
collect(min=n, max=n) | index(0)
```
where we only need to emit once the list is at size `n`.